external/llvm-project/mlir/lib/Dialect/Tosa/Transforms/TosaPartition.cpp

```diff
@@ -539,3 +623,92 @@ class TosaPartitionPass : public TosaPartitionBase<TosaPartitionPass> {
 std::unique_ptr<Pass> mlir::tosa::createTosaPartitionPass() {
   return std::make_unique<TosaPartitionPass>();
 }
+
+std::unique_ptr<Pass>
+mlir::tosa::createTosaPartitionPass(mlir::tosa::PartitionConfig *config) {
+  return std::make_unique<TosaPartitionPass>(config);
+}
+
+namespace {
+
+class TestTosaPartitionOptionsPass
```

Comment:
New options-test pass, to be able to test partition configurations specified in the program instead of on the command line.  It defines some arbitrary sub-options that lead to setting up various configurations, using the same test file that the command-line-option testing does.